docs/source/developers/java/building.rst

```diff
@@ -0,0 +1,298 @@
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. highlight:: console
+
+.. _building-arrow-java:
+
+===================
+Building Arrow Java
+===================
+
+.. contents::
+
+System setup
+============
+
+Arrow Java uses the `Maven <https://maven.apache.org/>`_ build system.
+
+Building requires:
+
+* JRE 8, 9, 10, 11
+* Maven 3+
+
+Building
+========
+
+All the instructions below assume that you have cloned the Arrow git
+repository:
+
+.. code-block::
+
+    $ git clone https://github.com/apache/arrow.git
+    $ cd arrow
+    $ git submodule update --init --recursive
+
+Basic Installation
+------------------
+
+To build the default modules, go to the project root and execute:
+
+.. code-block::
+
+    $ cd arrow/java
+    $ export JAVA_HOME=<absolute path to your java home>
+    $ java --version
+    $ mvn clean install
+    [INFO] Reactor Summary for Apache Arrow Java Root POM 8.0.0-SNAPSHOT:
+    [INFO]
+    [INFO] Apache Arrow Java Root POM ......................... SUCCESS [  4.922 s]
+    [INFO] Arrow Format ....................................... SUCCESS [  4.395 s]
+    [INFO] Arrow Memory ....................................... SUCCESS [  0.938 s]
+    [INFO] Arrow Memory - Core ................................ SUCCESS [ 11.491 s]
+    [INFO] Arrow Memory - Unsafe .............................. SUCCESS [  5.449 s]
+    [INFO] Arrow Memory - Netty ............................... SUCCESS [  8.754 s]
+    [INFO] Arrow Vectors ...................................... SUCCESS [01:37 min]
+    [INFO] Arrow Compression .................................. SUCCESS [  5.907 s]
+    [INFO] Arrow Tools ........................................ SUCCESS [ 11.015 s]
+    [INFO] Arrow JDBC Adapter ................................. SUCCESS [ 13.751 s]
+    [INFO] Arrow Plasma Client ................................ SUCCESS [  3.934 s]
+    [INFO] Arrow Flight ....................................... SUCCESS [  0.649 s]
+    [INFO] Arrow Flight Core .................................. SUCCESS [ 38.331 s]
+    [INFO] Arrow Flight GRPC .................................. SUCCESS [  5.894 s]
+    [INFO] Arrow Flight SQL ................................... SUCCESS [ 21.862 s]
+    [INFO] Arrow Flight Integration Tests ..................... SUCCESS [  7.638 s]
+    [INFO] Arrow AVRO Adapter ................................. SUCCESS [ 14.447 s]
+    [INFO] Arrow Algorithms ................................... SUCCESS [ 35.105 s]
+    [INFO] Arrow Performance Benchmarks ....................... SUCCESS [  5.323 s]
+    [INFO] ------------------------------------------------------------------------
+    [INFO] BUILD SUCCESS
+    [INFO] ------------------------------------------------------------------------\
+
+Building JNI Libraries on Linux
+-------------------------------
+
+For Build C Data Interface lib & Build C++ Libs. Consider: Next step requires: Docker, Docker Compose, `archery`_.
```

Comment:
Build -> building? Actually, I'm a little confused by this line, are you trying to say to build the C/C++, it requires Docker etc?

Reply:
I guess we can be more explicit. "First, we need to build the C++ shared libraries that the JNI bindings will use. We can build these manually (link to C++ documentation here), or we can use Archery to build them using a Docker container. This will require installing Docker, Docker Compose, and Archery."

Reply:
Thanks, changed